### PR TITLE
Support for complex trigger action values

### PR DESCRIPTION
**Pull request by jlsalmon**
This PR builds on top of the currently unmerged #140, adding support for CRUD operations
on trigger actions with complex values (such as JSON payloads) commonly used in integrations
with e.g. Zapier, PagerDuty, JIRA etc.

The following now works:

trigger.tf:
```
resource "zendesk_trigger" "json-target" {
    title = "notification_target trigger with JSON payload"

    // ...

    action {
        field = "notification_target"
        value = jsonencode([
            "1234"
            data.template_file.payload.rendered
        ])
    }
}

data "template_file" "payload" {
    template = file("${path.module}/trigger_payload.json.tpl")
    vars = {
        // ...
    }
}
```

TODO:
- [ ] add tests

### CODE REVIEW

**Comment by tamccall** [comment]
Hi @jlsalmon looks like your are having trouble with the master build.

I think that build has been removed from master at this point so if you merge that into your pr it will probably resolve itself.

**Comment by jlsalmon** [comment]
Thanks @tamccall, I've rebased the original branch on to master. I had to update `go.mod` to get the builds to pass - apologies if I've not done this correctly, I'm not super familiar with golang modules yet.

File: go.mod [hunk 1]
```diff
@@ -1,32 +1,16 @@
...
-	github.com/hashicorp/terraform v0.11.14
-	github.com/hashicorp/yamux v0.0.0-20181012175058-2f1d1f20f75d // indirect
-	github.com/mattn/go-isatty v0.0.8 // indirect
-	github.com/mitchellh/cli v1.0.0 // indirect
-	github.com/mitchellh/copystructure v1.0.0 // indirect
-	github.com/mitchellh/go-homedir v1.1.0 // indirect
-	github.com/mitchellh/go-wordwrap v1.0.0 // indirect
-	github.com/mitchellh/hashstructure v1.0.0 // indirect
+	github.com/hashicorp/terraform v0.12.11
+	github.com/hashicorp/terraform-plugin-sdk v1.2.0
```
Comment (nukosuke) [inline go.mod]: It's fine here, but maybe we should replace all "github.com/hashicorp/terraform" to "github.com/hashicorp/terraform-plugin-sdk" in future.

https://www.terraform.io/docs/extend/plugin-sdk.html

> As of September 2019, Terraform provider developers importing the Go module github.com/hashicorp/terraform, known as Terraform Core, should switch to github.com/hashicorp/terraform-plugin-sdk, the Terraform Plugin SDK, instead.

**Comment by nukosuke** [comment]
@jlsalmon 
Thank you for taking over the feature!
LGTM 👍 

@tamccall 
upgrading v0.11 -> v0.12 would probably break the acceptance tests. 
But, I think that it's about time to support v0.12 and drop v0.11 if necessary.
How do you feel about that?

File: go.mod (hunk 1 above)
Comment (jlsalmon) [reply go.mod]: +1 - I'll wrap that up in a separate PR.

Comment (jlsalmon) [reply go.mod]: See #169

**Comment by nukosuke** [comment]
@jlsalmon 
I created new branch `terraform-v0.12` to aggregate all changes for supporting v0.12, 
and rebased your PRs to it.

**Pull request merged**